pkg/controller/submariner/submariner_controller.go

```diff
@@ -297,7 +297,9 @@ func (r *ReconcileSubmariner) reconcileServiceDiscovery(submariner *submopv1a1.S
 					ClusterID:                submariner.Spec.ClusterID,
 					Namespace:                submariner.Spec.Namespace,
 					GlobalnetEnabled:         submariner.Spec.GlobalCIDR != "",
-					CustomDomains:            submariner.Spec.CustomDomains,
+				}
+				if submariner.Spec.CustomDomains != nil && len(submariner.Spec.CustomDomains) > 0 {
```

Comment:
`len(nil)` is 0, so

```suggestion
				if len(submariner.Spec.CustomDomains) > 0 {
```

(unlike Java, this won’t throw an NPE).